### PR TITLE
Feature/edgarfiling api

### PR DESCRIPTION
**Pull request by eharley19**

### CODE REVIEW

File: python/edgar_filing_searcher/models.py [hunk 1]
```diff
@@ -24,6 +24,10 @@ def __repr__(self):
 @dataclass
 class EdgarFiling(db.Model):
     """Define EdgarFiling Table"""
+    accession_no: str
+    cik_no: str
+    filing_date: db.Date
```
Comment (datdo) [inline python/edgar_filing_searcher/models.py:29]: db.Date vs datetime.date

File: python/edgar_filing_searcher/api/routes/edgarfiling.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+from edgar_filing_searcher.models import EdgarFiling
+
+filing_blueprint = Blueprint('edgarfiling', __name__)
+
+
+@filing_blueprint.route('/edgarfiling/<company_id>')
+def get_filings(company_id):
+    """Route for results for search by company id"""
+    date_format = '%Y-%m-%d'
+    start_date = request.args.get('start_date')
```
Comment (datdo) [inline python/edgar_filing_searcher/api/routes/edgarfiling.py:13]: these should be optional. So you should check to see if they exist first, and then apply the filter if they do.

**Review by datdo — CHANGES_REQUESTED** [review]

File: python/edgar_filing_searcher/api/routes/company.py [hunk 3]
```diff
@@ -1,12 +1,36 @@
...
-    companies = Company.query.filter(Company.cik_no.like(f"{company_id}%"))
-    return jsonify(list(companies))
+    date_format = '%Y-%m-%d'
+    start_date = request.args.get('start_date')
+    end_date = request.args.get('end_date')
+    filings = EdgarFiling.query.filter(EdgarFiling.cik_no.like(f"{company_id}%"))
+
+    if start_date:
+        filings = filings.filter(
+            EdgarFiling.filing_date >= datetime.strptime(start_date, date_format))
```
Comment (TheNorthEestern) [inline python/edgar_filing_searcher/api/routes/company.py:31]:
```suggestion
        filings = filings.filter(
            EdgarFiling.filing_date >= datetime.strptime(start_date, date_format)
        )
```

File: python/edgar_filing_searcher/api/routes/company.py [hunk 4]
```diff
@@ -1,12 +1,36 @@
...
+    if end_date:
+        filings = filings.filter(
+            EdgarFiling.filing_date <= datetime.strptime(end_date, date_format))
```
Comment (TheNorthEestern) [inline python/edgar_filing_searcher/api/routes/company.py:34]:
```suggestion
        filings = filings.filter(
            EdgarFiling.filing_date <= datetime.strptime(end_date, date_format)
        )
```

**Review by TheNorthEestern — APPROVED** [review]

**Review by datdo — APPROVED** [review]

**Pull request merged**